### PR TITLE
Multitenant OIDC support added to the documentation

### LINKED ISSUE #5723: OIDC Multi-tenancy Support Documentation
I made a PR review comment for https://github.com/helidon-io/helidon/pull/5168 but it seems to have gone unaddressed. Issue https://github.com/helidon-io/helidon/issues/4936.

This is a notable big feature and needs documentation.  I'd also even suggest we write a blog on this feature as well.

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #5723

### CODE REVIEW

**Comment by Verdent** [comment]
Hi @Geetha-Savithriamma , may I please ask you for a review? :-)

File: docs/includes/security/providers/oidc.adoc [hunk 1]
```diff
@@ -79,6 +79,63 @@ introspection endpoint depending on configuration
...
+
+Example: Host header value from inbound request is `my.helidon.com` -> domain level 3 is `my`,
+domain level 2 is `helidon` and domain level 1 is `com`
+|[source, yaml]
+----
+tenant-id-domain-level: <domain level>
+----
+
+|`token-handler`
+|Tenant name information is expected to be provided via configured custom header value
```
Comment (Geetha-Savithriamma) [inline docs/includes/security/providers/oidc.adoc:119]: The tenant name information is expected to be provided through the configured custom header value. (replaced via with through and inserted a period)

File: docs/includes/security/providers/oidc.adoc [hunk 2]
```diff
@@ -79,6 +79,63 @@ introspection endpoint depending on configuration
...
+|[source, yaml]
+----
+tenant-id-handler:
+  header: "my-custom-header"
+----
+
+|`none`|No tenant name finding is used. Default tenant name `@default` is used instead|
```
Comment (Geetha-Savithriamma) [inline docs/includes/security/providers/oidc.adoc:126]: Please insert a period after instead.

File: docs/includes/security/providers/oidc.adoc [hunk 3]
```diff
@@ -79,6 +79,63 @@ introspection endpoint depending on configuration
...
+|===
+
+It is also possible to implement custom way of discovering the tenant name and tenant configuration.
```
Comment (Geetha-Savithriamma) [inline docs/includes/security/providers/oidc.adoc:129]: You can also implement a custom way of discovering the tenant name and tenant configuration. The custom tenant name discovery from request can be done by implementing SPI:

File: docs/includes/security/providers/oidc.adoc [hunk 4]
```diff
@@ -79,6 +79,63 @@ introspection endpoint depending on configuration
...
+Custom tenant name discovery from request can be done via implementing SPI:
+
+`io.helidon.security.providers.oidc.common.spi.TenantIdProvider`
+
+and custom tenant configuration discovery can be provided via implementing SPI:
```
Comment (Geetha-Savithriamma) [inline docs/includes/security/providers/oidc.adoc:134]: and the custom tenant configuration discovery can be provided by implementing SPI:

**Review by Geetha-Savithriamma — COMMENTED** [review]
Hello David,

I have provided a few suggestions. Do take a look and let me know if you require any clarifications.

Warm regards,
Geetha.

File: docs/includes/security/providers/oidc.adoc [hunk 5]
```diff
@@ -79,6 +79,63 @@ introspection endpoint depending on configuration
...
+----
+
+There are 4 ways how to provide required tenant information to Helidon by default.
+
+.Possible `tenant-id-style` configuration options
+[cols="2,4a,3a"]
+|===
+|key |description |additional config options
+
+|`host-header` |Tenant configuration will be selected based on your host present in the Host header value |{nbsp}
```
[the remark this answers is not in the document]
Comment (Verdent) [reply docs/includes/security/providers/oidc.adoc:105]: > Should we capitalize host as in the original sentence?

Yes, it is the name of the header. I changed it a bit, so it is better readable.

**Comment by Verdent** [comment]
@ljamen @Geetha-Savithriamma Is everything OK now? :-)

**Review by ljamen — APPROVED** [review]
Geetha provided a review and LGTM.

**Pull request merged**